packages/discovery/src/discovery/DiscoveryLogger.ts

```diff
@@ -1,16 +1,34 @@
-import { EthereumAddress } from '@l2beat/shared'
+import { EthereumAddress, Logger } from '@l2beat/shared'
 import chalk from 'chalk'
 
 interface LoggerOptions {
   enabled: boolean
 }
 
 export class DiscoveryLogger {
-  constructor(private readonly options: LoggerOptions) {}
+  private serverLogs = ''
 
-  static SILENT = new DiscoveryLogger({ enabled: false })
+  constructor(
+    private readonly logger: Logger,
+    private readonly options: LoggerOptions,
+  ) {}
+
+  static SILENT = new DiscoveryLogger(Logger.SILENT, { enabled: false })
+  static CLI = new DiscoveryLogger(Logger.SILENT, { enabled: true })
+  static SERVER = (logger: Logger) =>
+    new DiscoveryLogger(logger, { enabled: false })
```

Comment:
What about a slightly different solution? I dislike the coupling of the two loggers.
Instead I suggest two flags: enabled and buffered.
If enabled you console log.
If buffered you concat to `this.bufferedLogs`.
Both can be enabled or both can be disabled too.

Then the function `flush` just returns the buffer and clears it.
In the discovery then you use the regular logger to log the flushed messages.